pinot-segment-local/src/main/java/org/apache/pinot/segment/local/segment/creator/impl/SegmentColumnarIndexCreator.java

```diff
@@ -386,30 +386,78 @@ public void indexRow(GenericRow row)
       // bloom filter
       BloomFilterCreator bloomFilterCreator = _bloomFilterCreatorMap.get(columnName);
       if (bloomFilterCreator != null) {
-        bloomFilterCreator.add(columnValueToIndex.toString());
+        if (fieldSpec.isSingleValueField()) {
+          bloomFilterCreator.add(columnValueToIndex.toString());
+        } else {
+          Object[] values = (Object[]) columnValueToIndex;
+          for (Object value: values) {
+            bloomFilterCreator.add(value.toString());
+          }
+        }
       }
 
       // range index
       CombinedInvertedIndexCreator combinedInvertedIndexCreator = _rangeIndexFilterCreatorMap.get(columnName);
       if (combinedInvertedIndexCreator != null) {
         if (dictionaryCreator != null) {
```

Comment:
Suggest for checking SV then dictionary. MV is always dictionary encoded:
```
if (fieldSpec.isSingleValueField()) {
  if (dictionaryCreator != null) {
    ...
  } else {
    switch ...
  }
} else {
  assert dictionaryCreator != null;
  ...
}

Reply:
I was following the segment loading part, and there seems to be handling for both single and multi value for non dictionary encoded columns.. Should be adding this constraint in the offline path?

https://github.com/apache/pinot/blob/master/pinot-segment-local/src/main/java/org/apache/pinot/segment/local/segment/index/loader/invertedindex/RangeIndexHandler.java#L170

Reply:
Good point. Let's keep them consistent